File: src/ipc/history.ts
import { ipc } from './client';

export interface HistoryIndexEntry {
  id: string;
  timestamp: string;
  operation: string;
  summary: string;
  fileCount: number;
}

export const listHistory = (limit = 50) => ipc<HistoryIndexEntry[]>('list_history', { limit });
export const rollback = (id: string) => ipc<void>('rollback', { entryId: id });

File: src-tauri/src/commands/history.rs
use crate::error::Result;
use crate::history::{self, HistoryIndexEntry};

#[tauri::command]
pub fn list_history(limit: usize) -> Result<Vec<HistoryIndexEntry>> {
    let mut entries = history::read_index()?;
    entries.truncate(limit);
    Ok(entries)
}

#[tauri::command]
pub fn rollback(entry_id: String) -> Result<()> {
    history::rollback(&entry_id)
}
